gnd/src/main/java/com/google/android/gnd/persistence/local/room/TileEntity.java

```diff
@@ -0,0 +1,93 @@
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.annotation.NonNull;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.PrimaryKey;
+
+import com.google.android.gnd.model.tile.Tile;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+
+@AutoValue
+@Entity(tableName = "tile")
+public abstract class TileEntity {
+    @CopyAnnotations
+    @NonNull
+    @PrimaryKey
+    @ColumnInfo(name = "id")
+    public abstract String getId();
+
+    @CopyAnnotations
+    @NonNull
+    @ColumnInfo(name = "path")
+    public abstract String getPath();
+
+    @CopyAnnotations
+    @NonNull
+    @ColumnInfo(name = "state")
+    public abstract TileEntityState getState();
+
+    public static Tile toTile(TileEntity tileEntity) {
+        Tile.Builder tile =
+                Tile.newBuilder()
+                        .setId(tileEntity.getId())
+                        .setPath(tileEntity.getPath())
+                        .setState(toTileState(tileEntity.getState()));
+        return tile.build();
+    }
+
+    private static Tile.State toTileState(TileEntityState state) {
```

Comment:
I really wish there was a better way to define and map between these representations. Maybe there's something we can do in the future with https://kotlinlang.org/docs/reference/data-classes.html

[just sharing, no action required]

Reply:
Agreed! Are we planning on rolling our own support for that, or are we going to switch to Kotlin :D?